### ISSUE TITLE
[mingw64] Vulkan-Loader install making the whole build fail

### ISSUE THREAD

**Issue by homeofthebadguys** [m-ab-s/media-autobuild_suite#1484]
``` bash
media-autobuild_suite (64bit)18:43:52 
	Starting 64bit compilation of all tools
Imported options from ffmpeg_options.txt
License: nonfree
Imported options from mpv_options.txt
media-autobuild_suite (64bit): compiling global tools18:44:00 
	Starting 64bit compilation of global tools
 ├ Running rustup_update...
 ├ Running set_default_toolchain...
   Running git update for sccache...
 sccache git  ...................................... [Up-to-date]
   Running git update for ripgrep...
 ripgrep git  ...................................... [Up-to-date]
   Running git update for jo...
 jo git  ........................................... [Up-to-date]
   Running git update for jq...
 jq git  ........................................... [Up-to-date]
   Running git update for dssim...
 dssim git  ........................................ [Up-to-date]
   Running git update for freetype2...

 freetype2 git  .................................... [Up-to-date]
   Running git update for fontconfig...
 fontconfig git  ................................... [Up-to-date]
   Running git update for harfbuzz...
 harfbuzz git  ..................................... [Up-to-date]
   Running git update for fribidi...
 fribidi git  ...................................... [Up-to-date]
   Running git update for libass...
 libass git  ....................................... [Up-to-date]
 gnutls 3.6.10  .................................... [Up-to-date]
 libidn2 2.2.0  .................................... [Up-to-date]
 libpsl 0.21.0  .................................... [Up-to-date]
   Running git update for curl...
 curl git  ......................................... [Up-to-date]
   Running git update for libtiff...
 libtiff git  ...................................... [Up-to-date]
   Running git update for libwebp...
 libwebp git  ...................................... [Up-to-date]
 ├ FFmpeg and related apps will depend on OpenCL.dll
   Running git update for OpenCL-Headers...
 OpenCL-Headers git  ............................... [Up-to-date]
 rubberband 1.8.1  ................................. [Up-to-date]
   Running git update for zimg...
 zimg git  ......................................... [Up-to-date]
media-autobuild_suite (64bit): compiling audio tools18:49:42 
	Starting 64bit compilation of audio tools
 libilbc 2.0.3-dev  ................................ [Up-to-date]
   Running git update for flac...
 flac git  ......................................... [Up-to-date]
 vo-amrwbenc 0.1.3  ................................ [Up-to-date]
   Running git update for fdk-aac...
 fdk-aac git  ...................................... [Up-to-date]
   Running git update for opus...
 opus git  ......................................... [Up-to-date]
   Running git update for libsoxr...
 libsoxr git  ...................................... [Up-to-date]
   Running git update for codec2...
 codec2 git  ....................................... [Up-to-date]
 libgme 0.6.2  ..................................... [Up-to-date]
 libbs2b 3.1.0  .................................... [Up-to-date]
   Running git update for sndfile...
 sndfile git  ...................................... [Up-to-date]
 sox 14.4.2  ....................................... [Up-to-date]
   Running git update for openmpt...
 openmpt git  ...................................... [Up-to-date]
   Running git update for libmysofa...
 libmysofa git  .................................... [Up-to-date]
   Running git update for flite...
 flite git  ........................................ [Up-to-date]
 shine 3.1.1  ...................................... [Up-to-date]
media-autobuild_suite (64bit): compiling video tools18:50:13 
	Starting 64bit compilation of video tools
   Running git update for librtmp...
 librtmp git  ...................................... [Up-to-date]
   Running git update for vpx...
 vpx git  .......................................... [Up-to-date]
   Running git update for aom...
 aom git  .......................................... [Up-to-date]
   Running git update for dav1d...
 dav1d git  ........................................ [Up-to-date]
   Running git update for rav1e...
 rav1e git  ........................................ [Up-to-date]
   Running git update for kvazaar...
 kvazaar git  ...................................... [Up-to-date]
 sdl2 2.0.10  ...................................... [Up-to-date]
   Running git update for dvdread...
 dvdread git  ...................................... [Up-to-date]
   Running git update for dvdnav...
 dvdnav git  ....................................... [Up-to-date]
   Running git update for libaacs...
 libaacs git  ...................................... [Up-to-date]
   Running git update for libbdplus...
 libbdplus git  .................................... [Up-to-date]
   Running git update for libbluray...
 libbluray git  .................................... [Up-to-date]
 xavs 0.1  ......................................... [Up-to-date]
   Running git update for xavs2...
 xavs2 git  ........................................ [Up-to-date]
   Running git update for davs2...
 davs2 git  ........................................ [Up-to-date]
   Running git update for libzen...
 libzen git  ....................................... [Up-to-date]
   Running git update for libmediainfo...
 libmediainfo git  ................................. [Up-to-date]
   Running git update for mediainfo...
 mediainfo git  .................................... [Up-to-date]
 vidstab 1.10  ..................................... [Up-to-date]
 zvbi-0.2 0.2.35  .................................. [Up-to-date]
   Running git update for dlfcn-win32...
 dlfcn-win32 git  .................................. [Up-to-date]
   Running git update for frei0r...
 frei0r git  ....................................... [Up-to-date]
   Running git update for decklink-headers...
 decklink-headers git  ............................. [Up-to-date]
   Running git update for libmfx...
 libmfx git  ....................................... [Up-to-date]
   Running git update for AMF...
 AMF git  .......................................... [Up-to-date]
   Running git update for gpac...
 gpac git  ......................................... [Up-to-date]
   Running git update for SVT-HEVC...
 SVT-HEVC git  ..................................... [Up-to-date]
   Running git update for SVT-AV1...
 SVT-AV1 git  ...................................... [Up-to-date]
   Running git update for SVT-VP9...
 SVT-VP9 git  ...................................... [Up-to-date]
   Running git clone for xvc...
 ┌ xvc git  .................................. [Recently updated]
 ├ Running cmake...
 ├ Running build...
 ├ Running install...
 ├ Stripping...
 └ xvc-git  ........................................... [Updated]
   Running git update for x264...
 x264 git  ......................................... [Up-to-date]
   Running hg update for x265...
 x265 hg  .......................................... [Up-to-date]
   Running git update for ffnvcodec...
 ffnvcodec git  .................................... [Up-to-date]
   Running git update for srt...
 srt git  .......................................... [Up-to-date]
 python 3.7.5  ..................................... [Up-to-date]
 vapoursynth R48  .................................. [Up-to-date]
   Running git update for ffmpeg...
 ffmpeg git  ....................................... [Up-to-date]
   Running git update for angle...
 angle git  ........................................ [Up-to-date]
 ├ Compiling mpv with Vapoursynth R48
 ├ mpv will need vapoursynth.dll and vsscript.dll to use vapoursynth filter!
   Running git update for mujs...
 mujs git  ......................................... [Up-to-date]
   Running git update for vulkan-loader...
 ┌ vulkan-loader git  ........................ [Recently updated]
 ├ Running uninstall...
0001-fix-cross-compiling.patch
	Patch could not be applied with `git am`. Continuing without patching.
 ├ Running dependencies...
 ├ Installing Vulkan-Headers...
 ├ Running uninstall...
 ├ Running cmake...
 ├ Running build...
 ├ Running install...
 ├ Building Vulkan-Loader...
 ├ Running cmake...
 ├ Running build...
Likely error (tail of the failed operation logfile):
../loader/vk_loader_platform.h:373:13: warning: 'loader_platform_thread_init_cond' defined but not used [-Wunused-function]
  373 | static void loader_platform_thread_init_cond(loader_platform_thread_cond *pCond) { InitializeConditionVariable(pCond); }
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../loader/vk_loader_platform.h:364:34: warning: 'loader_platform_get_thread_id' defined but not used [-Wunused-function]
  364 | static loader_platform_thread_id loader_platform_get_thread_id() { return GetCurrentThreadId(); }
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../loader/vk_loader_platform.h:288:14: warning: 'loader_platform_basename' defined but not used [-Wunused-function]
  288 | static char *loader_platform_basename(char *pathname) {
      |              ^~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
build failed. Check C:/media-autobuild_suite/build/vulkan-loader-git/build-64bit/ab-suite.build.log
This is required for other packages, so this script will exit.
   Creating diagnostics file...
```
I have compiled this sometime ago, but just recently ran into some old issues, then this one.
I was just compiling for my machine (a 64-bit only build) when it started failing for first some old errors with documented solutions, then this recent one that wasn't documented here.  I have to wonder if this is all me, or is it another bug that I just ran into? 
[logs.zip](https://0x0.st/zIQy.zip)

**Comment by 1480c1** [comment]
```
0001-fix-cross-compiling.patch
	Patch could not be applied with `git am`. Continuing without patching.
```
@wiiaboo, it doesn't seem like the patch has been rebased.

**Comment by wiiaboo** [comment]
Either that or the patching code isn't working again with existing but outdated files.
It's erroring with undeclared `CM_GETIDLIST_FILTER_CLASS`, which is something that shinchiro's [patch](https://github.com/shinchiro/mpv-winbuild-cmake/blob/master/packages/vulkan-0001-cross-compile-static-linking-hacks.patch#L84) adds

**Comment by 1480c1** [comment]
We aren't using shinchiro's patch, we are using [DeadSix27's](https://github.com/DeadSix27/python_cross_compile_script/blob/master/patches/vulkan/0001-fix-cross-compiling.patch)

**Comment by wiiaboo** [comment]
So either ask him to rebase the patch or PR it.
Or just switch back to shinchiro's.

**Comment by 1480c1** [comment]
Right now testing shinchiro's patch, if it works, I will update mabs to use it instead for now

**Comment by wiiaboo** [comment]
Fixed with 6daf9253bdcdd154d121809cc77f942d6f629e39
